code/game/machinery/recycler.dm

```diff
@@ -123,7 +123,8 @@ var/const/SAFETY_COOLDOWN = 100
 			items_recycled++
 		else
 			playsound(loc, 'sound/machines/buzz-sigh.ogg', 50, 0)
-			AM.loc = loc
+			if(AM)
```

Comment:
This check should be higher up in the loop:
```
if(!AM || qdeleted(AM))
  continue
```
... though we really need to replace `qdeleted` with some cleaner macros.